erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -162,10 +139,10 @@ def get_stock_ledger_entries(self):
 		return sle_dict
 
 	def on_trash(self):
-		if self.status == 'Delivered':
-			frappe.throw(_("Delivered Serial No {0} cannot be deleted").format(self.name))
 		if self.warehouse:
 			frappe.throw(_("Cannot delete Serial No {0} in stock. First remove from stock, then delete.").format(self.name))
+		elif self.delivery_document_no:
+			frappe.throw(_("Delivered Serial No {0} cannot be deleted").format(self.name))
 
```

Comment:
Instead, shouldn't we check if there is any Stock Ledger Entry for this serial no?